boards/arm/96b_carbon_nrf51/96b_carbon_nrf51-pinctrl.dtsi

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&pinctrl {
+	uart0_default: uart0_default {
+		pins1 {
+			pins = <NRF_PIN(0, 29, UART_TX)>,
+			       <NRF_PIN(0, 11, UART_RX)>;
```

Comment:
another naming choice could be:
```
psels = <NRF_PSEL(0, 29, UART_TX)>,
        <NRF_PSEL(0, 11, UART_RX)>;
```

@carlescufi @anangl @mbolivar-nordic

Reply:
> `psels = <NRF_PSEL(0, 29, UART_TX)>,`

Interesting. If we are going to do it like this, maybe it would be worthwhile to change the order of the arguments so it looks like `NRF_PSEL(UART_TX, 0, 29)` (etc.) instead.

Just an idea because the product specification's document PSEL.TXD etc., so putting the pin function (TX) next to the `PSEL` token in DTS matches that order.

Reply:
Looks reasonable. I think the internal macros in `pinctrl_nrf.c` could also use different naming: `NRF_PSEL_UART` instead of `NRF_UART_PSEL`.

Reply:
I've renamed macros to follow `NRF_PSEL_XXX` naming

Reply:
So does that mean you're dropping the `psels` property idea?

Reply:
not yet, I'd like to hear opinions @carlescufi @anangl ?

Reply:
I thought I had already expressed mine above. I like this idea, @mbolivar-nordic's suggestion to change the order, too.

Reply:
Yes, I do like that idea. It is consistent with the datasheets and it describes better the purpose of the macro, which is to associate a function's signal to a pin number. Let us go with that then, really nice idea.

Reply:
updated PR